hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsNode.java

```diff
@@ -0,0 +1,247 @@
+package org.apache.mesos.hdfs.scheduler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.mesos.hdfs.config.HdfsFrameworkConfig;
+import org.apache.mesos.hdfs.state.IPersistentStateStore;
+import org.apache.mesos.hdfs.state.LiveState;
+import org.apache.mesos.hdfs.util.HDFSConstants;
+import org.apache.mesos.Protos.CommandInfo;
+import org.apache.mesos.Protos.Environment;
+import org.apache.mesos.Protos.ExecutorID;
+import org.apache.mesos.Protos.ExecutorInfo;
+import org.apache.mesos.Protos.Offer;
+import org.apache.mesos.Protos.OfferID;
+import org.apache.mesos.Protos.Resource;
+import org.apache.mesos.Protos.TaskID;
+import org.apache.mesos.Protos.TaskInfo;
+import org.apache.mesos.SchedulerDriver;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * HdfsNode base class. 
+ */
+public abstract class HdfsNode implements IOfferEvaluator, ILauncher {
+  private final Log log = LogFactory.getLog(HdfsNode.class);
+  private final LiveState liveState;
+  private final ResourceFactory resourceFactory;
+
+  protected final HdfsFrameworkConfig config;
+  protected final IPersistentStateStore persistenceStore;
+  protected final String name;
+
+  public HdfsNode(LiveState liveState, IPersistentStateStore persistentStore, HdfsFrameworkConfig config, String name) {
+    this.liveState = liveState;
+    this.persistenceStore = persistentStore;
+    this.config = config;
+    this.name = name;
+    this.resourceFactory = new ResourceFactory(config.getHdfsRole());
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  protected abstract String getExecutorName();
+  protected abstract List<String> getTaskTypes();
+
```

Comment:
This method (launch) should now be private. We are calling tryLaunch publicly but we are calling launch within the class itself only.